src/properties/components/Properties.tsx

```diff
@@ -50,28 +58,23 @@ export const Properties = () => {
     () => tryFindByUUID(uiSchema, selection?.uuid),
     [selection, uiSchema]
   );
-  const canSetUISchemaOptions = (
-    uiElement: EditorUISchemaElement | undefined,
-    updatedProperties: any
-  ): boolean =>
-    !!uiElement &&
-    !isEqual(updatedProperties, uiElement.options) &&
-    !(uiElement.options === undefined && isEmpty(updatedProperties));
 
   const updateProperties = useCallback(
     ({ data: updatedProperties }) => {
-      if (canSetUISchemaOptions(uiElement, updatedProperties)) {
+      if (canUpdateUiSchemaElement(uiElement, updatedProperties)) {
         dispatch(Actions.setUiSchemaOptions(uiElement, updatedProperties));
       }
     },
     [dispatch, uiElement]
   );
+
   if (!selection) return <NoSelection />;
 
   const properties = getProperties(uiElement, schema);
+  const data = cloneTree(uiElement);
```

Comment:
Do we need to clone here? JSON Forms should never modify the original data object